src/concrete_intersection.jl

```diff
@@ -654,3 +655,48 @@ end
 function intersection(L1::LinearMap{N}, L2::LinearMap{N}) where {N}
     return intersection(linear_map(L1.M, L1.X), linear_map(L2.M, L2.X))
 end
+
+"""
+function intersection(X::CartesianProductArray{N},
+                      Y::AbstractPolyhedron{N}) where {N}
+
+Return the intersection of CartesianProductArray and AbstractPolyhedron.
+
+### Input
+
+ - `X` -- Cartesian Product Array of convex sets
+ - `Y` -- AbstractPolyhedron
+
+### Output
+
+The Cartesian Product Array obtained by the intersection of the nonzero blocks of `X` and `Y`.
+
+### Notes
+
+This function computes an intersection of CartesianProductArray and AbstractPolyhedron only for blocks, which are not free in AbstractPolyhedron object.
+"""
+function intersection(X::CartesianProductArray{N},
+                      Y::AbstractPolyhedron{N}) where {N}
+
+    # preallocate the resulting CartesianProductArray with size hint
+    result = CartesianProductArray(length(X.array), N)
+
+    if isbounded(Y)
+        # no free variables
+        vars = Vector{Int}(1:dim(Y))
```

Comment:
why convert to vector? this allocates. just let `vars = 1:dim(Y)` should work?

also could you please change the `vars` to something more readable, like `constrained_vars`?

Reply:
Because `1:dim(Y)` return `Range`, while `constrained_dimensions` returns an array.